implementation/src/main/java/io/smallrye/metrics/elementdesc/adapter/cdi/CDIMemberInfoAdapter.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.smallrye.metrics.elementdesc.adapter.cdi;
+
+import io.smallrye.metrics.elementdesc.MemberInfo;
+import io.smallrye.metrics.elementdesc.adapter.MemberInfoAdapter;
+
+import java.lang.reflect.AnnotatedElement;
+import java.lang.reflect.Member;
+
+public class CDIMemberInfoAdapter<T extends Member & AnnotatedElement> implements MemberInfoAdapter<T> {
```

Comment:
what is the justification for having the MemberInfoAdapter interface?

Reply:
It's probably to have this in Quarkus: https://github.com/quarkusio/quarkus/pull/2695/files#diff-af549f20edb10b48235c11ee599888e3R17 right?

Reply:
Yes, it is an abstraction to provide a possibility to pass metadata about fields/methods/beans/injectionPoints without depending on any particular API, each runtime can provide its own implementation. Quarkus doesn't support CDI extensions so the metadata needs to be based off Jandex instead of `java.lang.reflect` classes. So Thorntail and WildFly for now will use the default `Raw` implementations, Quarkus will have its own based on Jandex.